src/components/NewEventForm/NewEventForm.tsx

```diff
@@ -0,0 +1,218 @@
+    imageUrl: rest.imageUrl || "",
+    imageAlt: rest.imageAlt || "",
+    eventStatus: rest.eventStatus || EVENT_STATUS_OPTIONS.Draft
+  };
+  
+  // Validation
+  const EventFormSchema = Yup.object().shape({
+    name: Yup
+      .string()
+      .max(100, "Event name is too long")
+      .required("Event name is required"),
+    eventType: Yup
+      .string()
+      .oneOf(EVENT_OPTIONS)
+      .required('Event type is required'),
+    date: Yup
+      .date()
+      .required('Event date is required'),
+    startTime: Yup
+      .string()
+      .required("Start time is required")
+      .test("Start Time", "Start time cannot be later than end time", function(startTime) {
+        const startTimeHour = parseInt(this.parent.startTime.toString().substring(0,2));
+        const endTimeHour = parseInt(this.parent.endTime.toString().substring(0,2));
+        return startTimeHour < endTimeHour;
+      }),
+    endTime: Yup
+      .string()
+      .required("End time is required")
+      /*
+      .test("End Time", "End time cannot be earlier than start time", function(endTime) {
+        const startTimeHour = parseInt(this.parent.startTime.toString().substring(0,2));
+        const endTimeHour = parseInt(this.parent.endTime.toString().substring(0,2));
+        return endTimeHour > startTimeHour;
+      }) */,
+    description: Yup
+      .string()
+      .max(5000, "Description text is too long")
+      .required('Description of event is required'),
+    location: Yup
+      .string()
+      .max(200, "Location text is too long"),
+    imageUrl: Yup
+      .string()
+      .url("URL not valid"),
+    eventStatus: Yup
+      .string()
+      .oneOf([EVENT_STATUS_OPTIONS.Draft, EVENT_STATUS_OPTIONS.Published])
+      .required('Event status is required')
+  });
+
+  return (
+    <div>
+      <Formik
+        initialValues={InitialValues}
+        validationSchema = {EventFormSchema}
+        onSubmit={async (values, { setSubmitting }) => {
+          await new Promise(r => {
+            setSubmitting(true);
+      
+            try {
+              onSubmit ({
+                id : rest.id,
+                name : values.name,
+                description : values.description,
+                startTime : parseDate(eventDate, values.startTime),
+                endTime : parseDate(eventDate, values.endTime),
+                imageUrl : values.imageUrl,
+                imageAlt : values.imageAlt,
+                location : values.location,
+                eventType : values.eventType,
+                eventStatus : values.eventStatus,
+                teamsUrl : values.teamsUrl,    
+              });    
+            } catch (error) {}
+          });
+          setSubmitting(false);
+        }}
+      >
+        <EventFormWrapper>
+          <h1>{rest.headerTitle}</h1>
+          <Form>
+            <FormikInputField
+              label = "Event name *"
+              name = "name"
+              type = "text"
```

Comment:
It's not normal to have a space between attributes/props name and value like this. You should use Prettier if you haven't already installed the plugin.
```suggestion
              label="Event name *"
              name="name"
              type="text"
```
There's more of these below